client/components/Settings.js

```diff
@@ -15,20 +15,48 @@ import CustomBankField from './CustomBankField';
 import {OpCatChartTypeSelect, OpCatChartPeriodSelect} from './Charts';
 import T from './Translated';
 
-
 class Account extends React.Component {
 
+    constructor(props) {
+        super(props);
+        this.state = {
+            isDefault: (store.getDefaultAccountId() === this.props.account.id)
```

Comment:
How about, instead of storing the "isDefault" member as part of the state of this component, you just compute it in `render`? That avoids duplicating state, which is always nice. (In the next refactorings, most components will be stateless)